pages/search.js

```diff
@@ -145,13 +145,11 @@ export default function Search({
     }
 
     const timer = setTimeout(() => {
-      router.replace(
-        {
-          pathname: "/search",
-          query: { userSearchParam: inputValue },
-        },
-        undefined,
-        { shallow: true },
+      const newUrl = `/search?userSearchParam=${inputValue}`;
+      window.history.replaceState(
```

Comment:
Is it not possible with react/next router? https://nextjs.org/docs/pages/building-your-application/routing/linking-and-navigating#shallow-routing

Reply:
I tried it, but didn't work as expected and currently we're using this method but page gets reloaded causing the mobile keyboard close and open during search

Reply:
My idea is update the url without page gets reloaded and it works fine in this case

Reply:
> Is it not possible with react/next router? https://nextjs.org/docs/pages/building-your-application/routing/linking-and-navigating#shallow-routing

According to nextjs docs `shallow-routing` is probably only relevant if our page uses `getStaticProps`, `getServerSideProps` or `getInitialProps` to avoid re-running these data fetching methods again when the url change. But unfortunately `next/router` still re-rendering the page. 

In my research, I came across this https://github.com/vercel/next.js/discussions/18072 discussion thread to solve this issue.